pandas/core/arrays/floating.py

```diff
@@ -0,0 +1,636 @@
+from pandas.compat.numpy import function as nv
+from pandas.util._decorators import cache_readonly
+
+from pandas.core.dtypes.cast import astype_nansafe
+from pandas.core.dtypes.common import (
+    is_bool_dtype,
+    is_datetime64_dtype,
+    is_float,
+    is_float_dtype,
+    is_integer,
+    is_integer_dtype,
+    is_list_like,
+    is_object_dtype,
+    pandas_dtype,
+)
+from pandas.core.dtypes.dtypes import register_extension_dtype
+from pandas.core.dtypes.missing import isna
+
+from pandas.core import nanops, ops
+from pandas.core.array_algos import masked_reductions
+from pandas.core.ops import invalid_comparison
+from pandas.core.ops.common import unpack_zerodim_and_defer
+from pandas.core.tools.numeric import to_numeric
+
+from .masked import BaseMaskedArray, BaseMaskedDtype
+
+if TYPE_CHECKING:
+    import pyarrow  # noqa: F401
+
+
+class FloatingDtype(BaseMaskedDtype):
+    """
+    An ExtensionDtype to hold a single size of floating dtype.
+
+    These specific implementations are subclasses of the non-public
+    FloatingDtype. For example we have Float32Dtype to represent float32.
+
+    The attributes name & type are set when these subclasses are created.
+    """
+
+    name: str
+    base = None
+    type: Type
+
+    def __repr__(self) -> str:
+        return f"{self.name}Dtype()"
+
+    @property
+    def _is_numeric(self) -> bool:
+        return True
+
+    @cache_readonly
+    def numpy_dtype(self) -> np.dtype:
+        """ Return an instance of our numpy dtype """
+        return np.dtype(self.type)
+
+    @cache_readonly
+    def kind(self) -> str:
+        return self.numpy_dtype.kind
+
+    @cache_readonly
+    def itemsize(self) -> int:
+        """ Return the number of bytes in this dtype """
+        return self.numpy_dtype.itemsize
+
+    @classmethod
+    def construct_array_type(cls) -> Type["FloatingArray"]:
+        """
+        Return the array type associated with this dtype.
+
+        Returns
+        -------
+        type
+        """
+        return FloatingArray
+
+    def _get_common_dtype(self, dtypes: List[DtypeObj]) -> Optional[DtypeObj]:
+        # for now only handle other floating types
+        if not all(isinstance(t, FloatingDtype) for t in dtypes):
+            return None
+        np_dtype = np.find_common_type(
+            [t.numpy_dtype for t in dtypes], []  # type: ignore[union-attr]
+        )
+        if np.issubdtype(np_dtype, np.floating):
+            return FLOAT_STR_TO_DTYPE[str(np_dtype)]
+        return None
+
+    def __from_arrow__(
+        self, array: Union["pyarrow.Array", "pyarrow.ChunkedArray"]
+    ) -> "FloatingArray":
+        """
+        Construct FloatingArray from pyarrow Array/ChunkedArray.
+        """
+        import pyarrow  # noqa: F811
+
+        from pandas.core.arrays._arrow_utils import pyarrow_array_to_numpy_and_mask
+
+        pyarrow_type = pyarrow.from_numpy_dtype(self.type)
+        if not array.type.equals(pyarrow_type):
```

Comment:
ideally also push to the super-class (looks doable if you use self.construct_array_type